src/components/learner-credit-management/data/utils.js

```diff
@@ -551,3 +553,59 @@ export const isLmsBudget = (
   activeIntegrationsLength,
   isUniversalGroup,
 ) => activeIntegrationsLength > 0 && isUniversalGroup;
+
+/**
+ * Determines if the course has already started. Mostly used around text formatting for tense
+ * Introduces 'jitter' in the form of a 30 second offset to take into account any additional
+ * formatting that takes place down stream related to setting values to today's date through dayjs()
+ *
+ * This should also help with reducing 'flaky' tests.
+ *
+ * @param start
+ * @returns {boolean}
+ */
+export const hasCourseStarted = (start) => dayjs(start).isBefore(dayjs().subtract(30, 'seconds'));
+
+/**
+ * Returns assignable course runs within the threshold of within the subsidies expiration date
+ * offset by the STALE_ENROLLMENT_DROPOFF_DAYS constant. It sorts it from the soonest expiring
+ * enroll-by date and the enroll-by date and upgrade deadline has been normalized to ISO format.
+ *
+ * @param courseRuns
+ * @param subsidyExpirationDatetime
+ * @returns {*}
+ */
+export const assignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime }) => {
+  const clonedCourseRuns = courseRuns.map(a => ({
+    ...a,
+    enrollBy: dayjs(a.enrollBy * 1000).toISOString(),
+    upgradeDeadline: dayjs(a.upgradeDeadline * 1000).toISOString(),
+  }));
+  const sortedCourseRuns = clonedCourseRuns.sort((a, b) => a.enrollBy - b.enrollBy);
+  const filteredCourseRuns = sortedCourseRuns.filter(
+    ({ enrollBy }) => dayjs(enrollBy).isBefore(
+      dayjs(subsidyExpirationDatetime).add(STALE_ENROLLMENT_DROPOFF_DAYS, 'days'),
+    ),
+  );
+  return filteredCourseRuns;
+};
+
+/**
+ * If the start date of the course is before today offset by the START_DATE_DEFAULT_TO_TODAY_THRESHOLD_DAYS
+ * then return today's formatted date. Otherwise, pass-through the start date in ISO format.
+ *
+ * For cases where a start date does not exist, just return today's date.
+ *
+ * @param start
+ * @param format
+ * @returns {string}
+ */
+export const setStaleCourseStartDates = ({ start }) => {
```

Comment:
[update] From our discussion related in scrum, we do indeed want parity between admin+learner.

* learner portal's fallback to original start date should also consider the "old start date" threshold also getting introduced in admin portal.
* admin portal should only consider using today's date for the start date when the course is self-paced.
* admin portal should only consider using today's date for the start date when there's enough time to complete as determined by the existing/similar logic implemented [here](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/bd4087e1370d8c97b32888514c5fa1d0636f3d88/src/components/course/data/utils.jsx#L57-L59).

Reply:
For total agreeement, this would be the goal of implementing the new `start` date threshold logic within the `hasTimeToComplete` conditional as an OR statement 
``` jsx
    if (isCourseSelfPaced(courseRun.pacingType)) {
      if (hasTimeToComplete(courseRun) || OLD_START_DATE_LOGIC) {
        // always today's date (incentives enrollment)
        return intl.formatMessage(messages.courseStartDate, {
          startDate: dayjs().format(DATE_FORMAT),
        });
      }
      return intl.formatMessage(messages.courseStartedDate, {
        startDate: dayjs(courseStartDate).format(DATE_FORMAT),
      });
    }
    return intl.formatMessage(messages.courseStartedDate, {
      startDate: dayjs(courseStartDate).format(DATE_FORMAT),
    });
```